samples/servlet-echo/pom.xml

```diff
@@ -0,0 +1,178 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.microsoft.bot.sample</groupId>
+    <artifactId>bot-servlet-echo</artifactId>
+    <packaging>war</packaging>
+    <version>1.0.0</version>
+
+    <name>${project.groupId}:${project.artifactId}</name>
+    <url>http://maven.apache.org</url>
+
+    <licenses>
+        <license>
+            <name>MIT License</name>
+            <url>http://www.opensource.org/licenses/mit-license.php</url>
+        </license>
+    </licenses>
+
+    <developers>
+        <developer>
+            <name>Bot Framework Development</name>
+            <email></email>
+            <organization>Microsoft</organization>
+            <organizationUrl>https://dev.botframework.com/</organizationUrl>
+        </developer>
+    </developers>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.deploy.skip>false</maven.deploy.skip>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>javax.servlet-api</artifactId>
+            <version>3.1.0</version>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.module</groupId>
+            <artifactId>jackson-module-parameter-names</artifactId>
+            <version>2.9.2</version>
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.datatype</groupId>
+            <artifactId>jackson-datatype-jdk8</artifactId>
+            <version>2.9.2</version>
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.datatype</groupId>
+            <artifactId>jackson-datatype-jsr310</artifactId>
+            <version>2.9.2</version>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
```

Comment:
>slf4j-api< [](start = 24, length = 10)

Random q: Is this being pulled in because of autorest?  I think this was the library that indirectly pulls in a strange version log4j that I was mentioning earlier..

Reply:
While you don't see it in this version, I took a different strategy in an upcoming PR where I deal with dependencies and logging.  Since MS Rest is using slf4j, and slf4j is a reasonable solution to the plethora of Java logging libraries, I removed log4j from the core and just use slf4j.  This moves the dependency issue out of our core SDK and into the bot.  slf4j-api doesn't have any dependencies, but slf4j-log4j12 does pull in the 1.2.x version of log4j, instead of the latest 2.x.  My understanding is that there are newer bindings for the Log4j 2 as there are differences between the 1.2.x version and the 2.x versions.

Reply:
Got it!  I just recall being locked into an older version of log4j somehow.  Makes sense!